client/request.go

```diff
@@ -129,6 +130,14 @@ func (cli *Client) sendClientRequest(ctx context.Context, method, path string, q
 			return serverResp, err
 		}
 
+		if nErr, ok := err.(*url.Error); ok {
+			if nErr, ok := nErr.Err.(*net.OpError); ok {
+				if os.IsPermission(nErr.Err) {
+					return serverResp, errors.Wrap(ErrPermissionDenied, cli.host)
```

Comment:
used this
`errors.Wrapf(err, "Got permission denied while trying to connect to the Docker daemon socket at %v", cli.host)`
and go this output when reproducing the error. Is this more what we want?
`Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.25/containers/json?all=1: dial unix /var/run/docker.sock: connect: permission denied`